paywall/src/data-iframe/start/makePurchaseCallback.js

```diff
@@ -0,0 +1,44 @@
+import { getLocks, getTransactions, getKeys } from '../cacheHandler'
+
+// extraTip is not implemented yet, we need to do bignumber math, so post-pone this until walletService adds support
+const makePurchaseCallback = ({
+  walletService,
+  web3Service,
+  requiredConfirmations,
+  update,
+  window,
+}) =>
+  async function purchase(lockAddress /*, extraTip */) {
+    const [
+      { purchaseKey, processKeyPurchaseTransactions },
+      locks,
+      startingTransactions,
+      keys,
+    ] = await Promise.all([
+      import('../blockchainHandler/purchaseKey'),
```

Comment:
Can we only use import at the top of the file for consistency with the rest of the app please?

Reply:
absolutely not! this is the most important implementation detail that makes the code splitting of the cache from the blockchain handler possible (see https://webpack.js.org/guides/code-splitting/#dynamic-imports).

I'll add a series of comments to the code since this is not an intuitive code pattern.